src/box/box.cc

```diff
@@ -4296,3 +4297,231 @@ box_read_ffi_enable(void)
 	if (--box_read_ffi_disable_count == 0)
 		box_read_ffi_is_disabled = false;
 }
+
+/**
+ * Drop the constraint or foreign key. If fieldno is equal to UINT32_MAX then
+ * the tuple constraint or tuple foreign key is dropped, otherwise the field
+ * constraint or field foreign key is dropped.
+ */
+static int
+constraint_drop(uint32_t space_id, const char *name, uint32_t name_len,
+		uint32_t fieldno, enum tuple_constraint_type type)
+{
+	const struct space *space = space_by_id(space_id);
+	uint32_t total_count;
+	struct tuple_constraint_def *cdefs;
+	if (fieldno != UINT32_MAX) {
+		assert(fieldno < space->def->field_count);
+		total_count = space->def->fields[fieldno].constraint_count;
+		cdefs = space->def->fields[fieldno].constraint_def;
+	} else {
+		total_count = space->def->opts.constraint_count;
+		cdefs = space->def->opts.constraint_def;
+	}
+	bool is_exists = false;
+	uint32_t count = 0;
+	for (uint32_t i = 0; i < total_count; ++i) {
+		if (cdefs[i].type != type)
+			continue;
+		++count;
+		uint32_t len = MIN(cdefs[i].name_len, name_len);
+		if (memcmp(cdefs[i].name, name, len) == 0)
+			is_exists = true;
+	}
+	if (!is_exists) {
+		int errcode = type == CONSTR_FKEY ? ER_NO_SUCH_FOREIGN_KEY :
+			      ER_NO_SUCH_CONSTRAINT;
+		diag_set(ClientError, errcode, tt_cstr(name, name_len),
+			 space_name(space));
+		return -1;
+	}
+	struct region *region = &fiber()->gc;
+	uint32_t used = region_used(region);
+	size_t size = 64 + name_len;
+	char *raw = (char *)region_alloc(region, size);
+	if (raw == NULL) {
+		diag_set(OutOfMemory, size, "region_alloc", "raw");
+		return -1;
+	}
+	char *pos = mp_encode_array(raw, 1);
```

Comment:
Just in case you missed my first comment, you can also use `mp_vformat()`, it might be significantly simpler than manual encoding and it is not like we are running for perf here. Up to you.

Reply:
Thank you for the review! I fixed the above bug. Also I refactored the code using `mp_vformat()`. I really missed that part of the comment.